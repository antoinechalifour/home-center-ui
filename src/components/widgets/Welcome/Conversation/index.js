import { graphql } from 'react-apollo'
import { conversation } from 'queries/conversation'
import Conversation from './Conversation'

const conversationOptions = {
  name: 'conversation'
}

export default graphql(conversation, conversationOptions)(Conversation)
