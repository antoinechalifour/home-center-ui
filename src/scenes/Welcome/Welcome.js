import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

// From: http://www.english-test.net/forum/sutra289476.html
const allConfigurations = {
  morning: {
    message: 'Good morning'
  },
  afternoon: {
    message: 'Good afternoon'
  },
  evening: {
    message: 'Good evening'
  },
  night: {
    message: 'Good night'
  }
}

export default function Welcome ({ name, date }) {
  const hours = date.getHours()
  let configuration

  if (hours < 4) {
    configuration = allConfigurations.night
  } else if (hours <= 12) {
    configuration = allConfigurations.morning
  } else if (hours <= 17) {
    configuration = allConfigurations.afternoon
  } else if (hours <= 22) {
    configuration = allConfigurations.evening
  } else {
    configuration = allConfigurations.night
  }

  return <Message>{configuration.message}, {name}</Message>
}

Welcome.proptypes = {
  name: PropTypes.string.isRequired,
  date: PropTypes.instanceOf(Date).isRequired
}

const Message = styled.div`
  text-align: center;
  font-size: 32px;
  padding: 24px;
`
