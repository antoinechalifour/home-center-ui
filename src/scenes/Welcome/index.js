import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

// From: http://www.english-test.net/forum/sutra289476.html
const allConfigurations = {
  morning: {
    message: 'Good morning'
  },
  afternoon: {
    message: 'Good afternoon'
  },
  evening: {
    message: 'Good evening'
  },
  night: {
    message: 'Good night'
  }
}

export default function Welcome ({ name, date }) {
  const hours = date.getHours()
  let configuration

  if (hours < 4) {
    configuration = allConfigurations.night
  } else if (hours <= 12) {
    configuration = allConfigurations.morning
  } else if (hours <= 17) {
    configuration = allConfigurations.afternoon
  } else if (hours <= 22) {
    configuration = allConfigurations.evening
  } else {
    configuration = allConfigurations.night
  }

  return (
    <Container>
      <Message>{configuration.message}, {name}</Message>
    </Container>
  )
}

const Container = styled.div`
  text-align: center;
  font-size: 32px;
  padding: 39px 0;
`

const Message = styled.div`
  position: relative;
  padding: 12px;

  &::before,
  &::after {
    content: '';
    display: block;
    position: absolute;
    left: 25%;
    width: 50%;
    height: 1px;
    background: #fff;
    opacity: .15;
  }

  &::before {
    top: 0;
  }

  &::after {
    bottom: 0;
  }
`
