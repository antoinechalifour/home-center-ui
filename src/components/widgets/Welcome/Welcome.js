import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Conversation from './Conversation'

// From: http://www.english-test.net/forum/sutra289476.html
const allConfigurations = {
  morning: {
    message: 'Good morning'
  },
  afternoon: {
    message: 'Good afternoon'
  },
  evening: {
    message: 'Good evening'
  },
  night: {
    message: 'Good night'
  }
}

export default function Welcome ({ name, date }) {
  const hours = date.getHours()
  let configuration

  if (hours < 4) {
    configuration = allConfigurations.night
  } else if (hours <= 12) {
    configuration = allConfigurations.morning
  } else if (hours <= 17) {
    configuration = allConfigurations.afternoon
  } else if (hours <= 22) {
    configuration = allConfigurations.evening
  } else {
    configuration = allConfigurations.night
  }

  return (
    <Message>
      <span>{configuration.message}, {name}</span>
      <Conversation />
    </Message>
  )
}

Welcome.proptypes = {
  name: PropTypes.string.isRequired,
  date: PropTypes.instanceOf(Date).isRequired
}

const Message = styled.div`
  padding: 24px 0;
  display: flex;
  align-items: center;

  > :first-child {
    text-align: center;
    font-size: 32px;
    flex: 1;
  }
`
