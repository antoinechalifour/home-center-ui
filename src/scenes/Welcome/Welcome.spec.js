import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import Welcome from '.'

const tests = [
  // annÃ©e, mois[, jour[, heures[, minutes[, secondes[, millisecondes]]]]]
  {
    type: 'night',
    date: new Date(2017, 12, 28, 3)
  },
  {
    type: 'morning',
    date: new Date(2017, 12, 28, 4)
  },
  {
    type: 'morning',
    date: new Date(2017, 12, 28, 12)
  },
  {
    type: 'afternoon',
    date: new Date(2017, 12, 28, 13)
  },
  {
    type: 'afternoon',
    date: new Date(2017, 12, 28, 16)
  },
  {
    type: 'afternoon',
    date: new Date(2017, 12, 28, 17)
  },
  {
    type: 'evening',
    date: new Date(2017, 12, 28, 18)
  },
  {
    type: 'evening',
    date: new Date(2017, 12, 28, 21)
  },
  {
    type: 'evening',
    date: new Date(2017, 12, 28, 22)
  },
  {
    type: 'night',
    date: new Date(2017, 12, 28, 23)
  }
]

describe('scenes/Welcome', () => {
  tests.forEach(({ type, date }) =>
    it(`Should render "${type}" (${date})`, () => {
      const props = {
        name: 'John',
        date
      }
      const tree = renderer.create(<Welcome {...props} />)

      expect(tree).toMatchSnapshot()
    })
  )
})
