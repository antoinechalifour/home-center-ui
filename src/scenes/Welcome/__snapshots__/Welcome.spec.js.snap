// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenes/Welcome Should render "afternoon" (Sun Jan 28 2018 13:00:00 GMT+0100 (CET)) 1`] = `
<div
  className="sc-bdVaJa hevVxG"
>
  <div
    className="sc-bwzfXH eySVhi"
  >
    Good afternoon
    , 
    John
  </div>
</div>
`;

exports[`scenes/Welcome Should render "afternoon" (Sun Jan 28 2018 16:00:00 GMT+0100 (CET)) 1`] = `
<div
  className="sc-bdVaJa hevVxG"
>
  <div
    className="sc-bwzfXH eySVhi"
  >
    Good afternoon
    , 
    John
  </div>
</div>
`;

exports[`scenes/Welcome Should render "afternoon" (Sun Jan 28 2018 17:00:00 GMT+0100 (CET)) 1`] = `
<div
  className="sc-bdVaJa hevVxG"
>
  <div
    className="sc-bwzfXH eySVhi"
  >
    Good afternoon
    , 
    John
  </div>
</div>
`;

exports[`scenes/Welcome Should render "evening" (Sun Jan 28 2018 18:00:00 GMT+0100 (CET)) 1`] = `
<div
  className="sc-bdVaJa hevVxG"
>
  <div
    className="sc-bwzfXH eySVhi"
  >
    Good evening
    , 
    John
  </div>
</div>
`;

exports[`scenes/Welcome Should render "evening" (Sun Jan 28 2018 21:00:00 GMT+0100 (CET)) 1`] = `
<div
  className="sc-bdVaJa hevVxG"
>
  <div
    className="sc-bwzfXH eySVhi"
  >
    Good evening
    , 
    John
  </div>
</div>
`;

exports[`scenes/Welcome Should render "evening" (Sun Jan 28 2018 22:00:00 GMT+0100 (CET)) 1`] = `
<div
  className="sc-bdVaJa hevVxG"
>
  <div
    className="sc-bwzfXH eySVhi"
  >
    Good evening
    , 
    John
  </div>
</div>
`;

exports[`scenes/Welcome Should render "morning" (Sun Jan 28 2018 04:00:00 GMT+0100 (CET)) 1`] = `
<div
  className="sc-bdVaJa hevVxG"
>
  <div
    className="sc-bwzfXH eySVhi"
  >
    Good morning
    , 
    John
  </div>
</div>
`;

exports[`scenes/Welcome Should render "morning" (Sun Jan 28 2018 12:00:00 GMT+0100 (CET)) 1`] = `
<div
  className="sc-bdVaJa hevVxG"
>
  <div
    className="sc-bwzfXH eySVhi"
  >
    Good morning
    , 
    John
  </div>
</div>
`;

exports[`scenes/Welcome Should render "night" (Sun Jan 28 2018 03:00:00 GMT+0100 (CET)) 1`] = `
<div
  className="sc-bdVaJa hevVxG"
>
  <div
    className="sc-bwzfXH eySVhi"
  >
    Good night
    , 
    John
  </div>
</div>
`;

exports[`scenes/Welcome Should render "night" (Sun Jan 28 2018 23:00:00 GMT+0100 (CET)) 1`] = `
<div
  className="sc-bdVaJa hevVxG"
>
  <div
    className="sc-bwzfXH eySVhi"
  >
    Good night
    , 
    John
  </div>
</div>
`;
